### PR TITLE
Refactors & Removed some SQL injections

### PR DESCRIPTION
**Pull request by gumbarros**

### CODE REVIEW

File: src/JJMasterData.Commons/Dao/DataAccess.cs [hunk 1]
```diff
@@ -111,41 +112,18 @@ public bool KeepConnAlive
...
-    /// Initializes the class based on the name of the connection string
-    /// </summary>
-    /// <param name="connectionStringName">Connection String name configured in app.config</param>
+    
     public DataAccess(string connectionStringName)
     {
         ConnectionString = JJService.Settings.GetConnectionString(connectionStringName);
         ConnectionProvider = JJService.Settings.GetConnectionProvider(connectionStringName);
     }
-
```
Comment (gumbarros) [inline src/JJMasterData.Commons/Dao/DataAccess.cs:135]: Sorry, this comment was important, I will undo the delete

**Comment by gumbarros** [comment]
@LucioPelinson  ready for review

File: src/JJMasterData.Commons/Dao/DataAccess.cs [hunk 2]
```diff
@@ -282,33 +245,33 @@ public DataTable GetDataTable(DataAccessCommand dataAccessCommand)
...
-                foreach (DataAccessParameter parm in dataAccessCommand.Parameters)
+                foreach (var parm in dataAccessCommand.Parameters)
                 {
-                    DbParameter oPar = GetFactory().CreateParameter();
-                    oPar.DbType = parm.Type;
-                    oPar.Value = parm.Value ?? DBNull.Value;
-                    oPar.ParameterName = parm.Name;
-                    oPar.Direction = parm.Direction;
-                    cmd.Parameters.Add(oPar);
+                    var dbParameter = GetFactory().CreateParameter();
```
Comment (LucioPelinson) [inline src/JJMasterData.Commons/Dao/DataAccess.cs]: how about create a method that return a new parameter?

File: src/JJMasterData.Commons/Dao/DataAccess.cs [hunk 3]
```diff
@@ -1245,125 +1069,95 @@ public async Task<Hashtable> GetFieldsAsync(DataAccessCommand cmd)
...
-    ///Autor: Lucio Pelinson 14-04-2012
-    ///</remarks>
-    private string ParseSql(string sql)
+    [Obsolete("Impossible to be in DataAccess, needs to be in Provider")]
+    private DataAccessCommand GetTableExistsCommand(string table)
     {
-        return sql;
+        ///TODO: Migrate this to each provider.
+        const string sql = @"SELECT 
+            HAS_PERMS_BY_NAME
```
Comment (LucioPelinson) [inline src/JJMasterData.Commons/Dao/DataAccess.cs]: In SQLServer you need a special permission to execute this

File: src/JJMasterData.Commons/Dao/DataAccessProvider.cs [hunk 4]
```diff
@@ -7,6 +7,15 @@ namespace JJMasterData.Commons.Dao;
 
 public static class DataAccessProvider
 {
+    public const string Oracle = "System.Data.OracleClient";
```
Comment (LucioPelinson) [inline src/JJMasterData.Commons/Dao/DataAccessProvider.cs:10]: Great

File: src/JJMasterData.Commons/Util/StringManager.cs [hunk 5]
```diff
@@ -9,19 +9,19 @@ namespace JJMasterData.Commons.Util;
...
+    /// Author: Lucio Pelinson 21-05-2012
     /// </remarks>
-    public static string ClearText(string str)
+    public static string ClearText(string input)
     {
-        string sRet = str.Replace("'", "`");
-        sRet = sRet.Replace("<", "[");
-        sRet = sRet.Replace(">", "]");
-        sRet = sRet.Replace("--", "");
+        string result = input.Replace("'", "`");
```
Comment (LucioPelinson) [inline src/JJMasterData.Commons/Util/StringManager.cs:19]: this is better
```cs
result = result.Replace('<', '[');
```

than

```cs
result = result.Replace("<", "[");
```

File: src/JJMasterData.Commons/Dao/DataAccess.cs (hunk 2 above)
Comment (LucioPelinson) [inline src/JJMasterData.Commons/Dao/DataAccess.cs]: use CreateDbParameter method

**Pull request merged**